### PR TITLE
Update answers for existing user

### LINKED ISSUE #153: Backendin vastausten tallennus: Jos sähköposti löytyy tietokannasta, vastaukset päivitetään tietokantaan
#94
Pending review

### PR DESCRIPTION
**Pull request by Samipu**
Implemented update feature in backend/controllers/answers.js.
Added tests for the feature in answers.test.js.

Fixed linting issues in a couple frontend locations.

Branch renamed from original misleading name, thus history is not quite complete.

Ready for review.

### CODE REVIEW

File: backend/controllers/answers.js [hunk 1]
```diff
@@ -10,29 +10,43 @@ answersRouter.post('/', async (req, res) => {
 
   let userResult
 
-    try {
+  try {
+    if (email !== undefined) {
```
Comment (Nurou) [inline backend/controllers/answers.js]: if(email) is more concise and checks for null and undefined

File: backend/controllers/answers.js [hunk 2]
```diff
@@ -10,29 +10,43 @@ answersRouter.post('/', async (req, res) => {
...
+      let userId = -1
+      const existingUser = await User.findOne({
+        where: { email: email },
+      })
 
-      if(email !== undefined) {
+      if (existingUser !== null) {
```
Comment (Nurou) [inline backend/controllers/answers.js]: if(existingUser)

File: backend/controllers/answers.js [hunk 3]
```diff
@@ -10,29 +10,43 @@ answersRouter.post('/', async (req, res) => {
 
   let userResult
 
-    try {
+  try {
+    if (email !== undefined) {
+      let userId = -1
```
Comment (Nurou) [inline backend/controllers/answers.js]: why is this initialised to -1?

File: backend/controllers/answers.js [hunk 4]
```diff
@@ -10,29 +10,43 @@ answersRouter.post('/', async (req, res) => {
...
+        userId = user.id
+      }
+
+      const answersToQuestions = answers.map((answer) => ({
+        // eslint-disable-next-line node/no-unsupported-features/es-syntax
+        ...answer,
+        userId: userId,
+      }))
 
+      if (existingUser === null) {
```
Comment (Nurou) [inline backend/controllers/answers.js]: if(!existingUser)

File: backend/controllers/answers.js [hunk 5]
```diff
@@ -10,29 +10,43 @@ answersRouter.post('/', async (req, res) => {
...
-          ...answer,
-          userId: user.id,
-        }))
+        userId = user.id
+      }
+
+      const answersToQuestions = answers.map((answer) => ({
+        // eslint-disable-next-line node/no-unsupported-features/es-syntax
+        ...answer,
+        userId: userId,
```
Comment (Nurou) [inline backend/controllers/answers.js]: nitpick: could use object property shorthand: 

'userId: userId' would become 'userId'

See: https://eslint.org/docs/rules/object-shorthand

File: backend/tests/routes/answers.test.js [hunk 6]
```diff
@@ -80,6 +125,69 @@ describe('POST /api/answers', () => {
     expect(response.body.results.length).toEqual(4)
     done()
   })
+
+  it('User can submit answers with existing email', async (done) => {
```
Comment (Nurou) [inline backend/tests/routes/answers.test.js:127]: this tests probably has too many asserts. Could split up

File: backend/tests/routes/answers.test.js [hunk 7]
```diff
@@ -80,6 +125,69 @@ describe('POST /api/answers', () => {
...
+      email: 'testv2@gmail.com',
+      answers: testAnswers,
+    })
+
+    const response2 = await request(app).post('/api/answers').send({
+      email: 'testv2@gmail.com',
+      answers: testAnswers2,
+    })
+    expect(response2.status).toBe(200)
+    expect(response2.body.results).not.toEqual(null)
```
Comment (Nurou) [inline backend/tests/routes/answers.test.js]: should this be 'toNotEqual()'?

File: frontend/pages/survey/signup.jsx [hunk 8]
```diff
@@ -18,7 +18,7 @@ const Heading = styled.h3`
 const FormTitle = styled.h2`
   color: ${({ theme }) => theme.colors.blueDianne};
   font-family: Merriweather;
-  margin: 10px;
+  margin: 10px;s
```
Comment (Nurou) [inline frontend/pages/survey/signup.jsx]: typo?

**Review by Juustoisa — APPROVED** [review]
Looks good, great feedback by Nurou

**Pull request merged**